api/v1alpha1/validation/envoyproxy_validate.go

```diff
@@ -81,6 +85,21 @@ func validateProvider(spec *egv1a1.EnvoyProxySpec) []error {
 	return errs
 }
 
+func validateDeployment(spec *egv1a1.EnvoyProxySpec) []error {
+	var errs []error
+	if spec.Provider.Kubernetes != nil && spec.Provider.Kubernetes.EnvoyDeployment != nil {
```

Comment:
nit as follow up: can this be tackled via CEL ?

Reply:
Sure.